Source/Layout/ASLayout.mm

```diff
@@ -207,7 +215,10 @@ - (void)setRetainSublayoutLayoutElements:(BOOL)retainSublayoutLayoutElements
 
 - (ASLayout *)filteredNodeLayoutTree
 {
-  NSMutableArray *flattenedSublayouts = [NSMutableArray array];
+  if (ASLayoutIsFlattened(self)) {
+    self.retainSublayoutLayoutElements = YES;
```

Comment:
This seems like a strange side effect. Should we document why this needs to be the case?